extensions/mongodb/src/main/java/com/hazelcast/jet/mongodb/datalink/CloseableMongoClient.java

```diff
@@ -0,0 +1,168 @@
+/*
+ * Copyright 2023 Hazelcast Inc.
+ *
+ * Licensed under the Hazelcast Community License (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://hazelcast.com/hazelcast-community-license
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.hazelcast.jet.mongodb.datalink;
+
+import com.mongodb.ClientSessionOptions;
+import com.mongodb.client.ChangeStreamIterable;
+import com.mongodb.client.ClientSession;
+import com.mongodb.client.ListDatabasesIterable;
+import com.mongodb.client.MongoClient;
+import com.mongodb.client.MongoDatabase;
+import com.mongodb.client.MongoIterable;
+import com.mongodb.connection.ClusterDescription;
+import org.bson.Document;
+import org.bson.conversions.Bson;
+
+import javax.annotation.Nonnull;
+import java.util.List;
+
+/**
+ * MongoClient implementation that run given action on close and delegates all actions to some delegate.
+ */
+class CloseableMongoClient implements MongoClient {
+    final MongoClient delegate;
+    private final Runnable onClose;
+
+    CloseableMongoClient(MongoClient delegate, Runnable onClose) {
+        this.delegate = delegate;
+        this.onClose = onClose;
+    }
+
+    MongoClient unwrap() {
+        return delegate;
+    }
+
```

Comment:
Move the `close` method here and after that add a comment `All the following methods delegate to the actual client instance` or something similar